### ISSUE TITLE
Level 2 data browser not working

### ISSUE THREAD

**Issue by skymandr** [Odin-SMR/odin-api#44]
Från Donal:
> Noterade ocskå att plockningen i level 2 browser inte funkar. -  problem när alla data samlades ? 

http://odin.rss.chalmers.se/level2/ALL-Strat-v3.0.0/2/3455232065/

**Comment by skymandr** [comment]
@donal-mur We need more details before we can act on this. Can you document the problem?

**Comment by donal-mur** [comment]
if you look at say 2007-12-25 till 26 och försöka plota en level 2 så kommer det upp en blank sida med bara titel (körde Chrome)

**Comment by joakimmoller** [comment]
related to #43 ?

**Comment by local-minimum** [comment]
## The response JSON contains a lot of `null`:
In particular
```
"VMR":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null
```

Many (all?) "Quality" is `null`
And there's a case of a null `"Count"`, but that one might have been intentional for some reason.

 ## And a case of not a number
`MinLmFactor":NaN,`

So it looks like something

**Comment by local-minimum** [comment]
The VMR null list is created if the found VMR of the product is a scalar `NaN` or `None`. Compared to another date, then there's no `NaN` (which I think web browsers may have an issue handling) and there's just one Count and some Quality `null`:s

**Comment by skymandr** [comment]
Where is count defined? Is it used by the plot routine?

**Comment by local-minimum** [comment]
Count is just set to `None` always so that's a dead end.

However the entire odinapi may produce json that includes NaN, which has poor support. Depending on how the json is parsed it may just return an empty object. And then there's something wrong with the VMR

**Comment by local-minimum** [comment]
From the day with failing, apriori does exist at least:
```
"Apriori":[
            3.304974608020195e-05,
            2.55665705558178e-05,
            2.81480910324576e-05,
            3.031820603022283e-05,
            3.131019648461786e-05,
            3.179012960220554e-05
          
],
```

**Comment by skymandr** [comment]
That tells me nothing, unfortunately. May mean something to @donal-mur or @BengtRydberg, though.

**Comment by donal-mur** [comment]
Seems very few points - this id the ozone apriori?

Not sure which Scan you ar talking about


---------------------------------------------------------------
Donal Murtagh
Prof Global Environmental Measurements
Chalmers tekniska högskola
tel +46 31 772 56 51
donal.murtagh@chalmers.se<mailto:donal.murtagh@chalmers.se>



On 24 Jun 2020, at 11:56, Martin Zackrisson <notifications@github.com<mailto:notifications@github.com>> wrote:



From the day with failing, apriori does exist at least:

"Apriori":[
            3.304974608020195e-05,
            2.55665705558178e-05,
            2.81480910324576e-05,
            3.031820603022283e-05,
            3.131019648461786e-05,
            3.179012960220554e-05

],

**Comment by local-minimum** [comment]
Sorry it's quite hard to read the response json and figuring out what the data is.
The VMR that has all null is:

```
          "Product":"Temperature / 545 GHz / 15 to 65 km",
          "Quality":null,
          "ScanID":3455232065,
          "Temperature":[],
          "VMR":[]
```

`[]` just means it's an array, not that it's empty (they are not).

The apriori reportet along side  the VMR is:
```          "Apriori":[
            213.9061044523805,
            199.9682906805689,
            191.43398179127337,
            195.8780616558468,
            203.65179528223229,
            210.2434614595096,
            215.5681321436153,
            218.2463045713601,
            221.07163818075264,
            224.8525262012206,
            228.69678726300472,
            232.61129878306463,
            237.75570524436947,
            243.77780209077216,
            250.1293009651505,
            256.9348165593259,
            262.82948799244724,
            266.1736177225501,
            268.25163103890685,
            267.6980319748208,
            264.66530313014806,
            259.3481676685462,
            252.52801874940872,
            244.82234696310275,
            236.78031766661178,
            228.86394278508766,
            221.44410691995466,
            214.80715765368024,
            209.17357444734458,
            204.716654349065,
            201.64198849668446,
            200.15714999071372,
            200.30859917118096,
            201.9130637533068,
            204.73294125306995,
            208.42175608357815,
            212.59946591445697,
            216.79979815347417,
            220.39127408913114,
            222.5899229768599,
            222.61071032496667,
            220.36543766031474,
            216.42784239359364,
            211.45808796102543,
            206.0318577096947,
            200.58939144746626
          
],

```

**Comment by local-minimum** [comment]
The api request to get the data is http://odin.rss.chalmers.se/rest_api/v5/level2/ALL-Strat-v3.0.0/2/3455232065/

Plotting the temperature vector (top to bottom) gives the following profile:
![temp-profile](https://user-images.githubusercontent.com/8041100/85841822-f61c7f80-b79e-11ea-9fc7-9e912b4fe865.png)

@donal-mur I don't know the details of the data good enough to spot any strangeness to the apriori of the temperature product that got the null vector VMR . But except for the very sharp nick in the curve at the top (note y-axis could be inverted), it looks pretty similar to other days that has valid products.

So it looks to me as if the processing has failed in some way but didn't mange to detect it and posted the result as if it was valid.

**Comment by donal-mur** [comment]
It was the ozone Apriori I was worried about

---------------------------------------------------------------
Donal Murtagh
Prof Global Environmental Measurements
Chalmers tekniska högskola
tel +46 31 772 56 51
donal.murtagh@chalmers.se<mailto:donal.murtagh@chalmers.se>



On 26 Jun 2020, at 11:24, Martin Zackrisson <notifications@github.com<mailto:notifications@github.com>> wrote:



The api request to get the data is http://odin.rss.chalmers.se/rest_api/v5/level2/ALL-Strat-v3.0.0/2/3455232065/

Plotting the temperature vector (top to bottom) gives the following profile:
[temp-profile]<https://user-images.githubusercontent.com/8041100/85841822-f61c7f80-b79e-11ea-9fc7-9e912b4fe865.png>

@donal-mur<https://github.com/donal-mur> I don't know the details of the data good enough to spot any strangeness to the apriori of the temperature product that got the null vector VMR . But except for the very sharp nick in the curve at the top (note y-axis could be inverted), it looks pretty similar to other days that has valid products.

So it looks to me as if the processing has failed in some way but didn't mange to detect it and posted the result as if it was valid.

**Comment by local-minimum** [comment]
There seemed to be two ozone products so I included both:
![odin-o3](https://user-images.githubusercontent.com/8041100/85854866-8665bf00-b7b5-11ea-9c95-fe0c51c5c6b9.png)
![odin-o3-668](https://user-images.githubusercontent.com/8041100/85854868-88c81900-b7b5-11ea-81fb-586995b1f53d.png)

**Comment by donal-mur** [comment]
Jo och min känsla är att de är 10 gånger för stora plötligt

Donal

---------------------------------------------------------------
Donal Murtagh
Prof Global Environmental Measurements
Chalmers tekniska högskola
tel +46 31 772 56 51
donal.murtagh@chalmers.se<mailto:donal.murtagh@chalmers.se>



On 26 Jun 2020, at 14:01, Martin Zackrisson <notifications@github.com<mailto:notifications@github.com>> wrote:



There seemed to be two ozone products so I included both:
[odin-o3]<https://user-images.githubusercontent.com/8041100/85854866-8665bf00-b7b5-11ea-9c95-fe0c51c5c6b9.png>
[odin-o3-668]<https://user-images.githubusercontent.com/8041100/85854868-88c81900-b7b5-11ea-81fb-586995b1f53d.png>

**Comment by local-minimum** [comment]
Grafterna på sidan för dagen innan visar ju på apriori med helt andra magnituder (om inte tiopotenser döljs i graferna)
![Screenshot from 2020-06-26 15-01-28](https://user-images.githubusercontent.com/8041100/85859839-17d92f00-b7be-11ea-84fe-79c19af71e0b.png)
![Screenshot from 2020-06-26 15-01-16](https://user-images.githubusercontent.com/8041100/85859849-1ad41f80-b7be-11ea-969c-a9d2ff42224e.png)

**Comment by local-minimum** [comment]
(jag antar att mina har inverterat y-axeln)

**Comment by donal-mur** [comment]
Jo det tror jag du har nog tryck och inte hjöd-
Nu är jag lite förvirrad om vad ni tittar på?  Känns som vi bör ha ett möte
Donal
---------------------------------------------------------------
Donal Murtagh
Prof Global Environmental Measurements
Chalmers tekniska högskola
tel +46 31 772 56 51
donal.murtagh@chalmers.se<mailto:donal.murtagh@chalmers.se>



On 26 Jun 2020, at 15:03, Martin Zackrisson <notifications@github.com<mailto:notifications@github.com>> wrote:



(jag antar att mina har inverterat y-axeln)

**Comment by local-minimum** [comment]
Det är nog bara jag som inte kan atmosfärsfysik. Det jag främst kollar på just nu är egentligen bara vaför det inte blir några grafer på dagen med konstig data, inte så mycket varför datan är konstig. Men la till info här så att när vi ska försöka reda ut det så behöver man inte börja från början.

**Comment by donal-mur** [comment]
Ok
Så här är det . Om apriorin är väldig fel som jag tror det har blivit i december så funkar inte inversionen då den kommer att starta alltför långt från de rätta värden och aldrig konvergera - dessutom försöker metoden att push mot en felaktig a-priori .  Konvergera den inte så blir det ingen resultat och ingen graf tror jag.

Vad jag tror man behöver göra är att jämföra apriori för dagar (typ början på månaden) med dagar som inte fungerade typ den 25 - juldagen och reda ut varför det blir fel

Donal

---------------------------------------------------------------
Donal Murtagh
Prof Global Environmental Measurements
Chalmers tekniska högskola
tel +46 31 772 56 51
donal.murtagh@chalmers.se<mailto:donal.murtagh@chalmers.se>



On 26 Jun 2020, at 15:07, Martin Zackrisson <notifications@github.com<mailto:notifications@github.com>> wrote:



Det är nog bara jag som inte kan atmosfärsfysik. Det jag främst kollar på just nu är egentligen bara vaför det inte blir några grafer på dagen med konstig data, inte så mycket varför datan är konstig. Men la till info här så att när vi ska försöka reda ut det så behöver man inte börja från början.

**Comment by local-minimum** [comment]
The page now works: http://odin.rss.chalmers.se/level2/ALL-Strat-v3.0.0/2/3455232065/
We have another issue for the question about apriori #43
